docs/details/vision.dox

```diff
@@ -133,20 +133,70 @@ equal to the number of features contained in the query array.
 \defgroup cv_func_nearest_neighbour nearestNeighbour
 \ingroup featmatcher_mat
 
-\brief Nearest Neighbour
-
-Calculates nearest distances between two 2-dimensional arrays containing
-features based on the type of distance computation chosen. Currently, \ref
-AF_SAD (sum of absolute differences), \ref AF_SSD (sum of squared differences)
-and \ref AF_SHD (hamming distance) are supported.
-One of the arrays containing the training data and the other the
-query data. One of the dimensions of the both arrays must be equal among them,
-identifying the length of each feature. The other dimension indicates the
-total number of features in each of the training and query arrays. Two
-1-dimensional arrays are created as results, one containg the smallest N
-distances of the query array and another containing the indices of these
-distances in the training array. The resulting 1-dimensional arrays have length
-equal to the number of features contained in the query array.
+\brief Determine the nearest neighbouring points to a given set of points
+
+In this function, a "point" is simply a geometric point's coordinates in an
+n-dimensional space. The coordinates are described along dim0 (much like a
+traditional column vector). The \p train and \p query arrays then, are each
+expected to be a list of such coordinates, enumerated along dim1 (the usage of
+dim0 and dim1 here can also be switched, as explained later).
+
+This function then calculates which points in the \p train array are nearest to
+each point in the \p query array, based on the distance metric specified by \p
+dist_type: \ref AF_SAD (sum of absolute differences), \ref AF_SSD (sum of
+squared differences), or \ref AF_SHD (hamming distance).
+
+The number of nearest neighboring points returned is specified by \p n_dist,
+and the results are described in two output arrays:
```

Comment:
Hmm I think the awkwardness of this sentence stems from the fact that its subject is the number of the points, instead of the points itself. Maybe we should just restructure this sentence to make the actual points the subject.

Reply:
Well, I suggested the change because of the word `described` which doesn't fit the context.